cli/tests/functional/pom.xml

```diff
@@ -89,24 +101,51 @@
             <version>${version.lib.helidon}</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>io.helidon.build-tools.common</groupId>
+            <artifactId>helidon-build-common-test-utils</artifactId>
+            <version>${project.version}</version>
+            <scope>test</scope>
+        </dependency>
     </dependencies>
 
     <build>
         <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-jar-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>default-jar</id>
+                        <phase>test-compile</phase>
+                    </execution>
+                </executions>
+            </plugin>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-surefire-plugin</artifactId>
                 <configuration>
-                    <skipTests>true</skipTests>
                     <systemPropertyVariables>
-                        <!-- version should be upgraded to 3.0.0 once released-->
-                        <helidon.test.version>2.4.0</helidon.test.version>
+                        <cli.version>${project.version}</cli.version>
                         <helidon.executable.directory>${project.basedir}/../../impl</helidon.executable.directory>
-                        <helidon.cli.version>3.0.0-M1</helidon.cli.version>
-                        <helidon.plugin.version>${project.version}</helidon.plugin.version>
+                        <io.helidon.build.common.maven.url.localRepo>${user.home}/.m2/repository</io.helidon.build.common.maven.url.localRepo>
```

Comment:
`${user.home}/.m2/repository` assumes the default local repository.
I think you might be able to derive the local repository with code using class-loader lookups. E.g. lookup a class or resource from a JAR in the class-path, and derive the repository from the URI and the coordinates.